### PR TITLE
[ci] Make regex we use for git-cop more flexible

### PR DESCRIPTION
**Pull request by bgeuken**
* Allows tags that are not sorted alphabetically.
* Adds frontend tag as to valid tags list

### CODE REVIEW

**Comment by bgeuken** [comment]
@hennevogel Please review

File: dist/git-cop_configuration.yml [hunk 1]
```diff
@@ -50,6 +50,6 @@
   :enabled: true
   :severity: :error
   :whitelist:
-    - \A(?!\s)(\[api\])?(\[backend\])?(\[ci\])?(\[dist\])?(\[doc\])?(\[webui\])? [a-zA-Z]
+    - \A(?!\s)(?:\[api\]|\[backend\]|\[ci\]|\[dist\]|\[doc\]|\[frontend\]|\[webui\])+ [a-zA-Z]
```
Comment (hennevogel) [inline dist/git-cop_configuration.yml]: I think we need a new dev-team-rule: Always comment your regexes :-)

Comment (bgeuken) [inline dist/git-cop_configuration.yml]: @hennevogel added

**Comment by Ana06** [comment]
@bgeuken @hennevogel 

That the tags should be alphabetically ordered is documented in https://github.com/openSUSE/open-build-service/blob/master/CONTRIBUTING.md#how-to-write-proper-commit-messages, so it should also be remove there. Although I would prefer to enforce the tag to be ordered alphabetically... :cry: 

Also, in the same section of the contributing guide the frontend tag is not documented...

File: dist/git-cop_configuration.yml [hunk 2]
```diff
@@ -50,6 +50,11 @@
   :enabled: true
   :severity: :error
   :whitelist:
-    - \A(?!\s)(\[api\])?(\[backend\])?(\[ci\])?(\[dist\])?(\[doc\])?(\[webui\])? [a-zA-Z]
+    # Commit header is allowed to start with either
+    #   * api, backend, ci, dist, doc, frontend and webui tags in edgy brackets in any order
+    # or
+    #   * 'Update' or 'Upgrade' (needed by depfu gem)
+    # In both cases they should be followed by the actual commit message subject
+    - \A(?!\s)(?:Update|Upgrade|\[api\]|\[backend\]|\[ci\]|\[dist\]|\[doc\]|\[frontend\]|\[webui\])+ \S+
```
Comment (Ana06) [inline dist/git-cop_configuration.yml]: @bgeuken You are allowing repeating tags, and this regular expression can be simplified :thinking:

**Pull request merged**